platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/module/telemetry/MapClickEvent.java

```diff
@@ -0,0 +1,122 @@
+package com.mapbox.mapboxsdk.module.telemetry;
+
+import android.content.Context;
+import android.os.Parcel;
+import android.os.Parcelable;
+
+import com.google.gson.annotations.SerializedName;
+import com.mapbox.android.telemetry.Event;
+import com.mapbox.android.telemetry.TelemetryUtils;
+
+class MapClickEvent extends Event implements Parcelable {
+  private static final String MAP_CLICK = "map.click";
+
+  @SerializedName("event")
+  private final String event;
+  @SerializedName("created")
+  private String created;
+  @SerializedName("gesture")
+  private final String gesture;
+  @SerializedName("lat")
+  private double latitude;
```

Comment:
can we double check which fields can be made final? It seems that latitude and couple others are required and can be made final